tests/integration/test_host_resolver_fail_count/test_case.py

```diff
@@ -0,0 +1,103 @@
+"""Test Interserver responses on configured IP."""
+import pytest
+import time
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+
+node = cluster.add_instance(
+    "node",
+    main_configs=["configs/config.d/cluster.xml", "configs/config.d/s3.xml"],
+    with_minio=True,
+)
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+# The same value as in ClickHouse, this can't be confugured via config now
+DEFAULT_RESOLVE_TIME_HISTORY_SECONDS = 2*60
+
+
+def test_host_resolver(start_cluster):
+    minio_ip = cluster.get_instance_ip("minio1")
+
+    # drop DNS cache    
+    node.set_hosts([
+        (minio_ip, "minio1"),
+        (node.ip_address, "minio1"), # no answer on 9001 port on this IP
+        ])
+    
+    node.query("SYSTEM DROP DNS CACHE")
+    node.query("SYSTEM DROP CONNECTIONS CACHE")
+
+    node.query("""
+                CREATE TABLE test (key UInt32, value UInt32)
+                Engine=MergeTree()
+                ORDER BY key PARTITION BY key
+                SETTINGS storage_policy='s3'
+                """)
+
+    initial_fails = "0\n"
+    k = 0
+    limit = 100
+    while initial_fails == "0\n":
+        node.query(f"""
+                    INSERT INTO test VALUES (0,{k})
+                    """)
+        # HostResolver chooses IP randomly, so on single call can choose worked ID
+        initial_fails = node.query("SELECT value FROM system.events WHERE event LIKE 'AddressesMarkedAsFailed'")
+        k += 1
+        if k >= limit:
+            # Dead IP was not choosen for 100 iteration.
```

Comment:
I do not understan why there should be dead IP's?

If you want to write integration test that you need to write s3 mock here and request it. That mock could fail at accepting connections from time to time. But still I do not know how to assign several IP's at docker to one host name.

Reply:
In this case connection on the same IP sometimes works, sometimes not.
Current logic:
- if IP not answers, we mark it with `faled` and `fail_count`, sometimes clear `failed` to check again if it alive now
- if IP answered once, we clear all marks, and on next fail start from blank page again
Main purpose not to cover flapped network, but downtimes some IPs in DNS record for a long time (forever in my initial case).

Reply:
But that test is likely never faced two consecutive errors. Maybe just one some time by chance. Minio is pretty stable in integration tests. It is a challenge to make it works unstable. I do not see how you try to make it works unstable in that test. And I do not know how to do it either.

Reply:
For tests here is a cluster with two nodes - clickhouse `node` (say 192.168.1.1) and minio `minio1` (say 192.168.1.2)
With `node.set_hosts` call I created a fake DNS-record on clickhouse node, and in this fake record both IPs marked as `minio1` IPs. So clickhouse thinks that `minio1` has two IPs - 192.168.1.1 and 192.168.1.2. But on IP 192.168.1.1 noone lists port 9001. So when HostResolver chooses 192.168.1.2 - than connection woks, if 192.168.1.1 - clickhouse gets connection error.
And in old code without fix in 100 attempts clickhouse gets near 40 errors in average. With fix - maximum 1.

Reply:
My bad, I did not noticed `node.set_hosts`.
Than it is a good test! Thanks.